src/renate/updaters/model_updater.py

```diff
@@ -125,15 +127,37 @@ def _load_best_checkpoint_and_save(self, trainer: Trainer, pl_module: LightningM
         # Reload best state.
         learner_state_path = Path(defaults.learner_state_file(self._output_state_folder))
         if learner_state_path.exists():
-            self._model.load_state_dict(torch.load(defaults.model_file(self.dirpath)))
-            pl_module.load_state_dict(self._model, torch.load(learner_state_path)["state_dict"])
+            loaded_state = trainer.strategy.load_checkpoint(learner_state_path)
+            pl_module.on_load_checkpoint(loaded_state)
+            # This loads the state dict only if its not Deepspeed.
```

Comment:
Unsure what the comment refers to. Can you be more specific about which part is needed for Deepspeed?